src/tests/handlers/subjects/teacher.test.ts

```diff
@@ -0,0 +1,96 @@
+// GET
+import { handler } from "../../../handlers/subjects/teacher"
+import { CustomDynamoDB } from "../../../dynamodb/database";
+import responseHelper from "../../../helpers/responseHelper";
+import { validateToken } from "../../../helpers/validations";
+import * as userMock from "../../mocks/users.json"
+import * as subjectMock from "../../mocks/subjects.json"
+
+jest.mock('../../../dynamodb/database', () => {
+    return ({
+        CustomDynamoDB: jest.fn(()=> {
+            return {
+                query: jest.fn((teacherEmail)=> {
+                    return Promise.resolve(
+                        subjectMock.filter((subject) => subject.teacherEmail === teacherEmail)
+                    )                             
+                })
+            }
+        })
+    }) 
+})
+
+jest.mock('../../../helpers/validations',() => {
+    return({
+        validateToken: jest.fn((token) => {
+            return (token ? {
+                userType: userMock[2].userType,
+                userEmail: userMock[2].email
+            } : null)                  
+        })
+    })
+})
+
+describe("Teacher handler", () => {
+    describe("Fail test", () => {
+        it("Should fail when user token is not given", async () => {
+            const response = await handler({
+                headers: {
+                    Authorization: null
+                }
+            })
+            
+            const body = JSON.parse(response.body)
+            expect(body.message).toEqual("Fail Authenticating");
+            expect(response.statusCode).toEqual(403);      
+        })
+
+        it("Should fail when the teacher email is not given", async () => {
+            const response = await handler({
+                headers: {
+                    Authorization: "someGivenToken"
+                },
+                pathParameters: null ?? {
```

Comment:
This coalescense operator will always return the right handed object and is useless